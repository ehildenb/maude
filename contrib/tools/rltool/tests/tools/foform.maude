load ../../backend-lib.maude
load ../../theories/numbers.maude

fmod NAT-TEST is
  pr FVP-NAT-PRED .
  pr FVP-NUMBERS .
endfm

fmod TEST is
  pr FOFORMSIMPLIFY .
  pr FOFORMREDUCE .
  pr FOFORM-OPERATIONS .
  pr FOFORM-DEFINEDOPS .
  pr NNF .
  pr CNF .
  pr DNF .

  op testmod : -> Module .
  eq testmod = upModule('NAT-TEST,true) .

  ops f1 f1' f2 f2' f3 f3' f4 f4' f5 : -> QFForm .

  eq f1  = 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat] ?= '_monus_['Y:Nat,'Y:Nat] .
  eq f2  = '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat] .
  eq f3  = f1 /\ f2 .
  eq f4  = f3 ==> f1 .

  eq f1' = 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat] != '_monus_['Y:Nat,'Y:Nat] .
  eq f2' = '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat] .
  eq f3' = f1 /\ f2' .
  eq f4' = f3 ==> f1' .

  ops fA fB fC fD fE fG fH fI fJ : -> QFForm .
  eq fA = 'X:Nat ?= 'Y:Nat .
  eq fB = fA /\ fA .
  eq fC = fA \/ fA .
  eq fD = fA <==> fB .
  eq fE = fA /\ ~ fA .
  eq fG = tt /\ fC .
  eq fH = ff /\ fC .
  eq fI = tt \/ fC .
  eq fJ = ff \/ fC .
endfm

set show stats off .
set show timing off .
set show loop stats off .
set show loop timing off .

--- check that variable extraction works
red vars(mtForm) .
red vars(f1) .
red vars(f2) .
red vars(f3) .
red vars(f4) .

--- check well-formedness
red wellFormed(testmod,mtForm) .
red wellFormed(testmod,f1) .
red wellFormed(testmod,f2) .
red wellFormed(testmod,f3) .
red wellFormed(testmod,f4) .

--- check varsafe/full reduction works
red reduce(testmod,false,f1 ) .
red reduce(testmod,false,f2 ) .
red reduce(testmod,false,f3 ) .
red reduce(testmod,false,f4 ) .
red reduce(testmod,false,f1') .
red reduce(testmod,false,f2') .
red reduce(testmod,false,f3') .
red reduce(testmod,false,f4') .

red reduce(testmod,true, f1 ) .
red reduce(testmod,true, f2 ) .
red reduce(testmod,true, f3 ) .
red reduce(testmod,true, f4 ) .
red reduce(testmod,true, f1') .
red reduce(testmod,true, f2') .
red reduce(testmod,true, f3') .
red reduce(testmod,true, f4') .

--- check NNF/CNF/DNF
red toNNF(f1) .
red toNNF(f2) .
red toNNF(f3) .
red toNNF(f4) .
red toCNF(f1) .
red toCNF(f2) .
red toCNF(f3) .
red toCNF(f4) .
red toDNF(f1) .
red toDNF(f2) .
red toDNF(f3) .
red toDNF(f4) .

--- check varsafe/standard simplification
red varsafe-simplify(f1) .
red varsafe-simplify(f2) .
red varsafe-simplify(f3) .
red varsafe-simplify(f4) .

red varsafe-simplify(toNNF(f1)) .
red varsafe-simplify(toNNF(f2)) .
red varsafe-simplify(toNNF(f3)) .
red varsafe-simplify(toNNF(f4)) .

red simplify(f1) .
red simplify(f2) .
red simplify(f3) .
red simplify(f4) .
red simplify(f1') .
red simplify(f2') .
red simplify(f3') .
red simplify(f4') .

red simplify(toNNF(f1)) .
red simplify(toNNF(f2)) .
red simplify(toNNF(f3)) .
red simplify(toNNF(f4)) .
red simplify(toNNF(f1')) .
red simplify(toNNF(f2')) .
red simplify(toNNF(f3')) .
red simplify(toNNF(f4')) .

--- check second series of formulas against simplification
red varsafe-simplify(fA) .
red varsafe-simplify(fB) .
red varsafe-simplify(fC) .
red varsafe-simplify(fD) .
red varsafe-simplify(fE) .
red varsafe-simplify(fH) .
red varsafe-simplify(fI) .
red varsafe-simplify(fJ) .

red varsafe-simplify(toNNF(fA)) .
red varsafe-simplify(toNNF(fB)) .
red varsafe-simplify(toNNF(fC)) .
red varsafe-simplify(toNNF(fD)) .
red varsafe-simplify(toNNF(fE)) .
red varsafe-simplify(toNNF(fH)) .
red varsafe-simplify(toNNF(fI)) .
red varsafe-simplify(toNNF(fJ)) .

red simplify(fA) .
red simplify(fB) .
red simplify(fC) .
red simplify(fD) .
red simplify(fE) .
red simplify(fH) .
red simplify(fI) .
red simplify(fJ) .

red simplify(toNNF(fA)) .
red simplify(toNNF(fB)) .
red simplify(toNNF(fC)) .
red simplify(toNNF(fD)) .
red simplify(toNNF(fE)) .
red simplify(toNNF(fH)) .
red simplify(toNNF(fI)) .
red simplify(toNNF(fJ)) .

quit
