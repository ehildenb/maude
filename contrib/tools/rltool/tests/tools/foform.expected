==========================================
reduce in TEST : vars(mtForm) .
result EmptyTypeSet: (none).EmptyTypeSet
==========================================
reduce in TEST : vars(f1) .
result NeQidSet: 'X:Nat ; 'Y:Nat
==========================================
reduce in TEST : vars(f2) .
result NeQidSet: 'Y:Nat ; 'Z:Nat
==========================================
reduce in TEST : vars(f3) .
result NeQidSet: 'X:Nat ; 'Y:Nat ; 'Z:Nat
==========================================
reduce in TEST : vars(f4) .
result NeQidSet: 'X:Nat ; 'Y:Nat ; 'Z:Nat
==========================================
reduce in TEST : wellFormed(testmod, mtForm) .
result Bool: true
==========================================
reduce in TEST : wellFormed(testmod, f1) .
result Bool: true
==========================================
reduce in TEST : wellFormed(testmod, f2) .
result Bool: true
==========================================
reduce in TEST : wellFormed(testmod, f3) .
result Bool: true
==========================================
reduce in TEST : wellFormed(testmod, f4) .
result Bool: true
==========================================
reduce in TEST : reduce(testmod, false, f1) .
result PosEqAtom: '0.Nat ?= '_+_['1.NzNat,'1.NzNat,'X:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f2) .
result NegEqAtom: '_+_['1.NzNat,'Y:Nat] != '_+_['1.NzNat,'Y:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f3) .
result EqConj: '0.Nat ?= '_+_['1.NzNat,'1.NzNat,'X:Nat] /\ '_+_['1.NzNat,
    'Y:Nat] != '_+_['1.NzNat,'Y:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f4) .
result QFForm: ~ ('0.Nat ?= '_+_['1.NzNat,'1.NzNat,'X:Nat] /\ '_+_['1.NzNat,
    'Y:Nat] != '_+_['1.NzNat,'Y:Nat]) \/ '0.Nat ?= '_+_['1.NzNat,'1.NzNat,
    'X:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f1') .
result NegEqAtom: '0.Nat != '_+_['1.NzNat,'1.NzNat,'X:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f2') .
result PosEqAtom: '_+_['1.NzNat,'Y:Nat] ?= '_+_['1.NzNat,'Y:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f3') .
result PosEqConj: '0.Nat ?= '_+_['1.NzNat,'1.NzNat,'X:Nat] /\ '_+_['1.NzNat,
    'Y:Nat] ?= '_+_['1.NzNat,'Y:Nat]
==========================================
reduce in TEST : reduce(testmod, false, f4') .
result QFForm: ~ ('0.Nat ?= '_+_['1.NzNat,'1.NzNat,'X:Nat] /\ '_+_['1.NzNat,
    'Y:Nat] != '_+_['1.NzNat,'Y:Nat]) \/ '0.Nat != '_+_['1.NzNat,'1.NzNat,
    'X:Nat]
==========================================
reduce in TEST : reduce(testmod, true, f1) .
result FalseAtom: ff
==========================================
reduce in TEST : reduce(testmod, true, f2) .
result FalseAtom: ff
==========================================
reduce in TEST : reduce(testmod, true, f3) .
result ConstConj: ff /\ ff
==========================================
reduce in TEST : reduce(testmod, true, f4) .
result QFForm: ff \/ ~ (ff /\ ff)
==========================================
reduce in TEST : reduce(testmod, true, f1') .
result TrueAtom: tt
==========================================
reduce in TEST : reduce(testmod, true, f2') .
result TrueAtom: tt
==========================================
reduce in TEST : reduce(testmod, true, f3') .
result ConstConj: tt /\ ff
==========================================
reduce in TEST : reduce(testmod, true, f4') .
result QFForm: tt \/ ~ (ff /\ ff)
==========================================
reduce in TEST : toNNF(f1) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : toNNF(f2) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : toNNF(f3) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : toNNF(f4) .
result EqDisj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : toCNF(f1) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : toCNF(f2) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : toCNF(f3) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : toCNF(f4) .
result EqDisj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : toDNF(f1) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : toDNF(f2) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : toDNF(f3) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : toDNF(f4) .
result EqDisj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : varsafe-simplify(f1) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : varsafe-simplify(f2) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : varsafe-simplify(f3) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : varsafe-simplify(f4) .
result QFForm: ~ ('_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat] /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]) \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : varsafe-simplify(toNNF(f1)) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : varsafe-simplify(toNNF(f2)) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : varsafe-simplify(toNNF(f3)) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : varsafe-simplify(toNNF(f4)) .
result EqDisj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : simplify(f1) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : simplify(f2) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : simplify(f3) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : simplify(f4) .
result QFForm: ~ ('_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat] /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]) \/ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : simplify(f1') .
result NegEqAtom: '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : simplify(f2') .
result PosEqAtom: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : simplify(f3') .
result PosEqConj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] /\ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : simplify(f4') .
result QFForm: ~ ('_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat] /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]) \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : simplify(toNNF(f1)) .
result PosEqAtom: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : simplify(toNNF(f2)) .
result NegEqAtom: '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : simplify(toNNF(f3)) .
result EqConj: '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
    /\ '_+_['Y:Nat,'1.NzNat] != 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],'Z:Nat]
==========================================
reduce in TEST : simplify(toNNF(f4)) .
result TrueAtom: tt
==========================================
reduce in TEST : simplify(toNNF(f1')) .
result NegEqAtom: '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],
    '3.NzNat]
==========================================
reduce in TEST : simplify(toNNF(f2')) .
result PosEqAtom: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat]
==========================================
reduce in TEST : simplify(toNNF(f3')) .
result PosEqConj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] /\ '_monus_['Y:Nat,'Y:Nat] ?= 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : simplify(toNNF(f4')) .
result EqDisj: '_+_['Y:Nat,'1.NzNat] ?= 'sd['_+_['Z:Nat,'Y:Nat,'1.NzNat],
    'Z:Nat] \/ '_monus_['Y:Nat,'Y:Nat] != 'sd['_+_['X:Nat,'5.NzNat],'3.NzNat]
==========================================
reduce in TEST : varsafe-simplify(fA) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fB) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fC) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fD) .
result QFForm: ~ ('X:Nat ?= 'Y:Nat) \/ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fE) .
result QFForm: ~ ('X:Nat ?= 'Y:Nat) /\ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fH) .
result PosConj: ff /\ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fI) .
result PosDisj: tt \/ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(fJ) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fA)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fB)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fC)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fD)) .
result EqDisj: 'X:Nat ?= 'Y:Nat \/ 'X:Nat != 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fE)) .
result EqConj: 'X:Nat ?= 'Y:Nat /\ 'X:Nat != 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fH)) .
result PosConj: ff /\ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fI)) .
result PosDisj: tt \/ 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : varsafe-simplify(toNNF(fJ)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(fA) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(fB) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(fC) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(fD) .
result TrueAtom: tt
==========================================
reduce in TEST : simplify(fE) .
result FalseAtom: ff
==========================================
reduce in TEST : simplify(fH) .
result FalseAtom: ff
==========================================
reduce in TEST : simplify(fI) .
result TrueAtom: tt
==========================================
reduce in TEST : simplify(fJ) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(toNNF(fA)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(toNNF(fB)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(toNNF(fC)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
==========================================
reduce in TEST : simplify(toNNF(fD)) .
result TrueAtom: tt
==========================================
reduce in TEST : simplify(toNNF(fE)) .
result FalseAtom: ff
==========================================
reduce in TEST : simplify(toNNF(fH)) .
result FalseAtom: ff
==========================================
reduce in TEST : simplify(toNNF(fI)) .
result TrueAtom: tt
==========================================
reduce in TEST : simplify(toNNF(fJ)) .
result PosEqAtom: 'X:Nat ?= 'Y:Nat
Bye.
